Teatoller/studentRptCard
# #007  Integrate xlsx record and nodemailer

#### What does this PR do?
- Integrate data from the xlsx worksheet with the nodemailer application to be able to send emails to each student in the database.
#### Description of Task to be completed?
- [x] import mailer to send emails in index.js
- [x] import email template
- [x] import the dotenv application
- [x] iterate through each record in the table
- [x] dispatch email to each student in the template

#### How should this be manually tested?
1. Navigate to a folder where you wish to host the application and open the terminal or your choice editor.
2. Run `mkdir my_project` on the terminal.
3. Run-on terminal `cd my_project`
4. Run-on terminal `git clone https://github.com/Teatoller/studentRptCard.git`
5. cd `studentRptCard.git`
6. git checkout ft-integrate-xlsxdata-and-nodemailer-003`
7. Add a .env file and check format using the .env.example file provided.
8. Install dependencies by running  `npm install` on terminal.
9. Use [Google Help centre](https://support.google.com/accounts/answer/185833) to add `studentRptCard` to liist of gmail apps and also to generate a password string to use ith the **.env file**. (Note: without doing this you will not succesfully test out the application)
10. run `node index.js` on the terminal. You should see printed on the terminal confirmation of sent emails  as in  **Figure 1**. and **Fig:2** Sample email
11. Check the email address you associated ith the **.env**. The mail Daemons will be in your inbox as the email used are not actual emails. You ill be able to visualize also the email template with the student details.
#### Any background context you want to provide?
- n/a
#### What are the relevant pivotal tracker stories?
- n/a
#### Screenshots (if appropriate)
**Fig:1**
![Screenshot from 2019-11-02 12-35-24](https://user-images.githubusercontent.com/39020475/68069525-0b591f80-fd72-11e9-906c-6ab88a42f062.png)

**Fig:2**
![Screenshot from 2019-11-02 12-37-16](https://user-images.githubusercontent.com/39020475/68069526-12802d80-fd72-11e9-9135-e4fa6d1790d2.png)


#### Questions:
- n/a
